version: "3"

services:

  db:
    image: postgres:16
    restart: on-failure
    volumes:
      - pg_data:/var/lib/postgresql/data
    env_file:
      - .env
    expose:
      - "5432"
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U postgres" ] # если свой пользователь - pg_isready -d 'POSTGRES_DB' -U 'POSTGRES_USER'
      interval: 10s
      timeout: 5s
      retries: 5


  redis:
    image: redis:latest
    expose:
      - "6379"
    healthcheck:
      test: [ "CMD-SHELL", "redis-cli ping" ]
      interval: 10s
      timeout: 5s
      retries: 5


  library_app:
    build: .
    depends_on:
      db:
        condition: service_healthy
      redis:
        condition: service_healthy
    tty: true
    command: bash -c "python3 manage.py migrate && python3 manage.py collectstatic --noinput && python3 manage.py runserver 0.0.0.0:8000"
    ports:
      - "8000:8000"
    volumes:
      - .:/library_app
    env_file:
      - .env


  celery:
    build: .
    restart: on-failure
    depends_on:
      library_app:
        condition: service_started
      db:
        condition: service_healthy
      redis:
        condition: service_healthy
    tty: true
    command: celery -A config worker -l INFO
    volumes:
      - .:/library_app
    env_file:
      - .env


  celery-beat:
    build: .
    restart: on-failure
    depends_on:
      library_app:
        condition: service_started
      db:
        condition: service_healthy
      redis:
        condition: service_healthy
    tty: true
    command: celery -A config beat -l INFO
    volumes:
      - .:/library_app
    env_file:
      - .env

volumes:
  pg_data: